### DISCUSSION TITLE
How Depth can save in grayscale png file ?

### DISCUSSION THREAD

**Discussion by damavand1** [isl-org/Open3D#5815] [General]
hi

When we want to create an gray we shoud set R=G=B and we know (0,0,0)=Black and (255,255,255)=white
that means each pixel has only 256 option. now how can we can save 0-3000 mm (3001 option) in grayscale file ?
I checked you saved depth image in BedroomRGBDImages sample in grayscale png file .
why you choose png and how you save D in gray pixel?

![000000](https://user-images.githubusercontent.com/104978921/209620117-99596bb1-7340-41aa-ad9c-ce45683399d7.png)

**Comment by jerome-godbout-lychens** [comment]
def normalized_depth_image(depth):
    normalized_depth = (depth - depth.min()) / (depth.max() - depth.min()) * 255
    normalized_depth = normalized_depth.astype(np.uint8)
    return normalized_depth

def depth_captured(self, depth):
        o3d.io.write_image("depth.png", o3d.geometry.Image(
            normalized_depth_image(np.asarray(depth))
        ))

scene.render_to_depth_image(self.depth_captured)
